packages/features/insights/server/trpc-router.ts

```diff
@@ -685,9 +681,18 @@ export const insightsRouter = router({
       return [];
     }
 
-    const result: IResultTeamList[] = belongsToTeams.map((membership) => {
-      return { ...membership.team };
-    });
+    // Filter teams to only include those where the user has insights.read permission
+    // This properly handles both PBAC permissions and traditional role-based access
+    const teamsWithAccess = await Promise.all(
+      belongsToTeams.map(async (membership) => {
+        const hasAccess = await checkInsightsPermission(user.id, membership.team.id);
+        return hasAccess ? membership.team : null;
+      })
+    );
+
+    const result: IResultTeamList[] = teamsWithAccess
+      .filter((team): team is NonNullable<typeof team> => team !== null)
```

Comment:
I think we can use `getTeamIdsWithPermission` method in permission-service to avoid N+1 queries here